### PR TITLE
Big docs/README refactoring

### PR DESCRIPTION
**Pull request by wlach**
* Move most reference and cookbook documentation in README.md into
  the "reference" and "cookbook" sections of the generated documentation,
  respectively.
* Try to steer people to the generated docs inside the README.md (since
  it is now basically just a set of quickstart instructions)
* Provide a bit of guidance that this repository isn't great for 3rd
  party contributors in a new CONTRIBUTING.md.

### CODE REVIEW

**Comment by wlach** [comment]
Preview available here (for some reason the dataset docs aren't showing up, but I think that's an existing issue and doesn't really affect other aspects of this PR): https://deploy-preview-2527--bigquery-etl-dev.netlify.app/

I may wait until after the holiday weekend to ask for feedback/review, as this is a rather large change and people might have opinions on how I did things.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Contributing to bigquery-etl
```
Comment (jklukas) [inline CONTRIBUTING.md:1]: +1 to this CONTRIBUTING. It nicely captures the constraints.

File: docs/bqetl.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 The `bqetl` command-line tool aims to simplify working with the bigquery-etl repository by supporting common workflows, such as creating, validating and scheduling queries or adding new UDFs.
 
+Running some commands, for example to create or query tables, will [require access to Mozilla's GCP Account](https://docs.telemetry.mozilla.org/cookbooks/bigquery/access.html#bigquery-access-request).
```
Comment (jklukas) [inline docs/bqetl.md:5]:
```suggestion
Running some commands, for example to create or query tables, will [require Mozilla GCP access](https://docs.telemetry.mozilla.org/cookbooks/bigquery/access.html#bigquery-access-request).
```

File: tests/README.md [hunk 3]
```diff
@@ -1,173 +1 @@
...
-circleci build --job integration \
-  --env GOOGLE_PROJECT_ID=bigquery-etl-integration-test \
-  --env GCLOUD_SERVICE_KEY=$gcloud_service_key
-
-# to run all jobs
-circleci build \
-  --env GOOGLE_PROJECT_ID=bigquery-etl-integration-test \
-  --env GCLOUD_SERVICE_KEY=$gcloud_service_key
-```
+For information on how to run tests, see https://mozilla.github.io/bigquery-etl/cookbooks/testing.html
```
Comment (jklukas) [inline tests/README.md:1]:
```suggestion
For information on how to run tests, see https://mozilla.github.io/bigquery-etl/cookbooks/testing/
```

**Review by jklukas — APPROVED** [review]
Thanks for working on this. The README has been too bloated for a while, and I do think this will make it more likely for users to find info relevant to their needs.

**Pull request merged**